public/app/features/variables/query/QueryVariableEditor.tsx

```diff
@@ -115,7 +113,13 @@ export class QueryVariableEditorUnConnected extends PureComponent<Props, State>
 
   renderQueryEditor = () => {
     const { editor, variable } = this.props;
-    if (!editor.extended || !editor.extended.dataSource || !editor.extended.VariableQueryEditor) {
+
+    if (
+      !editor.extended ||
+      editor.extended.kind !== 'query' ||
```

Comment:
There is no kind right? Why does this work?

Reply:
It almost definitely doesn't work 😅. Just did new style in a few places to demo it.

this PR is all _very_ incomplete, still draft work-in-progress